src/easylink/pipeline_graph.py

```diff
@@ -105,26 +108,82 @@ def spark_is_required(self) -> bool:
         """
         return any([implementation.requires_spark for implementation in self.implementations])
 
-    def get_input_slot_attributes(self, node: str) -> dict[str, dict[str, str | list[str]]]:
-        """Gets all of a node's input slot attributes from edges.
+    def get_whether_embarrassingly_parallel(self, node: str) -> bool:
+        """Determines whether a node is to be run in an embarrassingly parallel way.
 
         Parameters
         ----------
         node
-            The node name to get input slot attributes for.
+            The node name to determine whether or not it is to be run in an
+            embarrassingly parallel way.
 
         Returns
         -------
-            A mapping of node name to input slot attributes.
+            A boolean indicating whether the node is to be run in an embarrassingly
+            parallel way.
+        """
+        return self.nodes[node]["implementation"].is_embarrassingly_parallel
+
+    def any_embarrassingly_parallel(self) -> bool:
+        """Checks if any node is to be run in an embarrassingly parallel way."""
+        are_any = any(
+            [
+                self.get_whether_embarrassingly_parallel(node)
+                for node in self.implementation_nodes
+            ]
+        )
+        return are_any
```

Comment:
```suggestion
        return any(
            [
                self.get_whether_embarrassingly_parallel(node)
                for node in self.implementation_nodes
            ]
        )
```

nit: I think this is easier to read without the variable

Reply:
good catch, just lingering from a previous breakpoint